utils/appliance/__init__.py

```diff
@@ -142,6 +148,36 @@ def uninstall_ipa_client(self):
         assert return_code != 0
 
 
+class ApplianceLoggerAdapter(logging.LoggerAdapter):
+    def process(self, msg, kwargs):
+        msg = msg.strip()
+        if msg.startswith('['):
+            spacer = ''
+        else:
+            spacer = ': '
+        return '[{}]{}{}'.format(self.extra['appliance_ip'], spacer, msg), kwargs
+
+
+class RESTLoggerAdapter(logging.LoggerAdapter):
+    def process(self, msg, kwargs):
+        return '[REST]: {}'.format(msg), kwargs
+
+
+class SSHLoggerAdapter(logging.LoggerAdapter):
+    def process(self, msg, kwargs):
+        return '[SSH]: {}'.format(msg), kwargs
+
+    def trace(self, msg, *args, **kwargs):
```

Comment:
i guess thats acceptable, we should take a look at moving ssh logging to a own logger